committee-steering/sig-governance.md

```diff
@@ -0,0 +1,533 @@
+
+#### Oncall
+
+In order to facilitate consistently passing tests and triage of issues,
+an oncall rotation must be setup containing SIG ~~Foo~~ members.  The
+purpose of this group is to proactively monitor signals for the health
+of the areas owned by the SIG, and to provide an escalation point for
+urgent or critical issues.  If no oncall group is chosen, this group
+will default to the technical lead group.
+
+**Note:** Oncall calendar infrastructure for th community is
+forthcoming.
+
+#### Test [SLO][slo]
+
+SIG ~~Foo~~ has an objective of fixing any broken test within 2 days
+either by addressing the underlying issue, or rolling back the breaking
+change.  The oncall member has the power to freeze code merges for
+SIG ~~Foo~~ areas if tests remain broken for more than 2 days.
+
+#### Flaky tests
+
+Flaky tests are defined at having >10% failure rate by job.
+Tests observed to
+be consistently flaky will be prioritized and fixed within 1 release
+cycle.  If no members of the SIG can commit to fixing the flaky
+test within a release single, then Tech leads have the power to
+stop accepting PRs for new features until staffing is committed.
+
+#### Test escalation
+
+If assets owned by SIG ~~Foo~~ are suspected to be breaking other
+SIGs tests or making their tests flaky, then members of those
+SIGs may send an email to the sig mail list and escalate to the test
+oncall member.  If the issue cannot be resolved, the issue may be
+escalated to technical leads.
+
+### Addressing operational or technical blocking issues
+
+If for some reason the SIG is unable to meet its responsibilities
+and this is negatively impacting either the project or SIG,
+then the project lead or technical lead group (whoever is
+responsible for the issue), may enforce that the SIG stop accepting
+contributions for new development, and only accept bug fixes
+and sustaining engineering until the issue has been resolved.
+If no resolution can be found, the issue may be escalated to the
+[Steering Committee][steering-commitee].
+
+---
+
+# FAQ
+
+## Why is it so long
+
+This document attempts to ensure the institutional knowledge
+of individuals managing SIGs today is well documented and available
+to all members in the SIG.  There are actually many responsibilities and
+processes that have developed as part of facilitating management
+of the project by a SIG comprised of a decentralized and diverse
+group of individuals.
+
+## Why so much structure
+
+This document intentionally leans toward more structure rather than
+less.  There are 2 reasons for this:
+
+- It is easy for SIGs to remove or edit sections in this
+  template to reduce structure, whereas it is harder to make up new
+  structure from scratch
+- The SIG governance questionnaire showed a consistent preference for more
+  structure over less.  (almost all answers were to have the same or
+  more amount of structure)
+
+## Why use mailing lists for tracking roles
```

Comment:
This is the thing that gives me the most heartburn.

Overall, we've been trying to align code structure with people structure.  That would suggest that OWNERS files would be the canonical place to store this stuff.  But the problems with that are that (a) we don't have email for all people in that and (b) there is no way to send mail to that group.

Problem with google groups: (a) no API I can see to get membership lists (b) no relationship back to github (c) membership can be hidden if not careful and (d) there is no history/audit log of who is added when by whom.

In my mind the *ideal* way to do this would be to (a) use OWNERS as canonical (b) have a database someplace of github->email (this could be semi-private?) and (c) auto-populate the groups based on that.

The big problem is that there is no API to do c! The API for that is only available to gsuite members.  We do have a gsuite account for k8s.io and we could use that but that would be a *huge* change to move all sig communications there.  But the good thing would be that we would have a super consistent and clear system.

What I suspect is going to happen is that both OWNERS and the mailing lists will be canonical.  *sigh*.

Reply:
Another issue with OWNERs files is that they are not all equal and SIGs don't map to a single root directory.  We could setup conventions to get around this, but I haven't found OWNERs files to be great for membership tracking.

Maybe OWNERs files could have GoogleGroups as first class references?

Reply:
I am open to any solution, but I haven't seen OWNERs files used well for anything related to managing a group.  They do a good job of providing access control on resources to a group or list of individuals, but not a whole lot else.

More examples of things that they aren't optimized for:

- Docs sharing works with groups, but doesn't work with OWNERs files 
- Invitations to calendar events work with groups, but don't work with OWNERs files
- Some responsibilities aren't tied to files - e.g. recording and uploading SIG meetings, 
ensuring tests are passing.

Reply:
;)  Works for me.  For now we will do the best we can with tools available to us.